ui/pages/Project/components/EditDatasetsButton.jsx

```diff
@@ -170,20 +171,33 @@ const PANES = [
 
 const IGV_ONLY_PANES = [PANES[1]]
 
-const EditDatasetsButton = React.memo(({ user }) => (
-  (user.isDataManager || user.isPm) ? (
-    <Modal
-      modalName={MODAL_NAME}
-      title="Datasets"
-      size="small"
-      trigger={<ButtonLink>Edit Datasets</ButtonLink>}
-    >
-      <Tab panes={user.isDataManager ? PANES : IGV_ONLY_PANES} />
-    </Modal>
-  ) : null
-))
+const EditDatasetsButton = React.memo(({ project, user }) => {
+  const showLoadWorkspaceData = !project.isAnalystProject && project.canEdit
+  const title = showLoadWorkspaceData ? 'Request Additional Data from Anvil Workspace' : 'Datasets'
+  const text = showLoadWorkspaceData ? 'Request Additional Data' : 'Edit Datasets'
+  return (
+    (user.isDataManager || user.isPm || showLoadWorkspaceData) ? (
+      <Modal
+        modalName={MODAL_NAME}
+        title={title}
```

Comment:
it would be more readable to define the variable inline here rather than as a constant above. Same for `text`. In geenral, if you are only using a variable once its more readable not to make it a separate variable